.changeset/small-ties-sort.md

```diff
@@ -0,0 +1,44 @@
+---
+'astro': major
+---
+
+Fixes attribute rendering for non-[boolean attributes](https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML) with boolean values. This matches proper attribute handling in browsers and allows you to read the attribute values consistently.
+
+Given this Astro code (where `allowfullscreen` is a boolean attribute):
+
+```astro
+<p allowfullscren={true}></p>
+<p allowfullscren={false}></p>
+
+<p unknown={true}></p>
+<p unknown={false}></p>
+
+<p data-foo={true}></p>
+<p data-foo={false}></p>
+```
```

Comment:
This looks great! Just checking since this is an Astro file and in the HTML-ish part, should the comments being in `<!-- -->` form instead? Or is the JS form the norm. I can update if so.

Thanks for the better names 😄